src/EventStore.Core/ProcessTracker.cs

```diff
@@ -0,0 +1,190 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Threading;
+using EventStore.Common.Utils;
+using EventStore.Core.Services.Monitoring.Stats;
+using EventStore.Core.Services.Monitoring.Utils;
+using EventStore.Core.Time;
+using EventStore.Native.Monitoring;
+using HostStat;
+using Serilog;
+
+namespace EventStore.Core;
+
+public class ProcessTracker {
+	private static readonly ILogger _log =
+		Serilog.Log.ForContext(Serilog.Core.Constants.SourceContextPropertyName, "PROCESS-TRACKER-LOGGER");
+
+	private IClock _clock;
+	private Instant _lastTime = Instant.FromSeconds(0);
+	private Process? _process;
+	private LoadAverages _loadAverages;
+	private EsDriveInfo? _esDriveInfo;
+	private DiskIo? _diskIo;
+
+
+	private readonly HostStat.HostStat _stats;
+	private readonly DateTime _startTime;
+	private readonly TimeSpan _startCpuUsage;
+	private readonly string _dbPath;
+	private readonly PerfCounterHelper _perfCounter;
+	private readonly EventCountersHelper _eventCountersHelper;
+
+	public ProcessTracker(IClock clock, string dbPath) {
+		_clock = clock;
+		_stats = new HostStat.HostStat();
+		_dbPath = dbPath;
+		_perfCounter = new PerfCounterHelper(_log);
+		_startTime = DateTime.Now;
+		_startCpuUsage = Process.GetCurrentProcess().TotalProcessorTime;
+		_eventCountersHelper = new EventCountersHelper(2_500);
+	}
+
+	private T Bookkeeping<T>(Func<T> func) {
+		var now = _clock.Now;
+		if (_lastTime != Instant.FromSeconds(0) && now.ElapsedSecondsSince(_lastTime) < 5)
+			return func();
+
+		_lastTime = now;
+
+		if (OS.IsUnix) {
+			_loadAverages = _stats.GetLoadAverages();
+		}
+
+		_esDriveInfo = EsDriveInfo.FromDirectory(_dbPath, _log);
```

Comment:
it would be good to include the DiskName as a dimension of the metric

the old stats dont seem to have this, but it would also be cool to get the drive stats for the index drive as well as for the log drive if it is different

Reply:
Done